### PR TITLE
fix to support backward compatibility for compile tf job

### PR DESCRIPTION
**Pull request by christophmluscher**
@michelwi does this fix the issue?

### CODE REVIEW

**Comment by christophmluscher** [comment]
@michelwi sry did not see yours

**Comment by christophmluscher** [comment]
> which is actually not the case if you start the graph compilation while on a machine with active GPU, or am I wrong there?

I think the default in the old setup would be set in the `returnn.config`??? for example `device ="gpu"`

I am not sure if the availability of a GPU further influences this

**Comment by michelwi** [comment]
This line 
https://github.com/rwth-i6/returnn/blob/2ec00d42baf5acdf0a6f4b341fd1e6c050945008/tools/compile_tf_graph.py#L1480
Indicates the (new) default if no flag is set is "cpu", which would proliferate to this job.

**Comment by christophmluscher** [comment]
ah I thought we were talking about the old default. extend doc with `from commit HASH_date forward default is: CPU`??

**Comment by JackTemaki** [comment]
> 

I never set this in the returnn config... so well anything can happen. Willi is right, for the "new" RETURNN it defaults to cpu, and for old Returnn it either takes the device specified in the config OR it depends on the availability on the GPU if nothing is specified.

**Comment by michelwi** [comment]
Ok, I will abandon the other pr and then we can put as docstring whatever we see fit.

File: returnn/compile.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ def __init__(
         :param int eval:
         :param int search:
         :param int log_verbosity: RETURNN log verbosity from 1 (least verbose) to 5 (most verbose)
-        :param str device: optimize graph for cpu or gpu
+        :param str|None device: optimize graph for cpu or gpu
```
Comment (JackTemaki) [inline returnn/compile.py:46]:
```suggestion
        :param str|None device: optimize graph for cpu or gpu. If `None`, defaults to cpu for current RETURNN. 
            For any RETURNN version before `cd4bc382`, the behavior will depend on the `device` entry in the
            `returnn_conig`, which will depend on the availability of the GPU if not defined at all. 
```

Comment (JackTemaki) [reply returnn/compile.py:46]: maybe like this?

Comment (michelwi) [reply returnn/compile.py:46]: > the `device` entry in the
            `returnn_conig`, which will depend on the availability of the GPU if not defined at all.

Well not the device entry depends on the availability. Therefore

```
            For any RETURNN version before `cd4bc382`, the behavior will depend on the `device` entry in the
            `returnn_conig`, or on the availability of a GPU on the execution host if not defined at all.
```

**Comment by JackTemaki** [comment]
Oh damn, I did not see that black failed? Was the test that much delayed?

File: returnn/compile.py [hunk 2]
```diff
@@ -92,12 +94,13 @@ def run(self):
...
             "--eval=%d" % self.eval,
             "--search=%d" % self.search,
-            "--device=%s" % self.device,
             "--verbosity=%d" % self.verbosity,
             "--output_file=%s" % self.out_graph.get_path(),
             "--output_file_model_params_list=model_params",
             "--output_file_state_vars_list=state_vars",
         ]
+        if self.device is not None:
+          args.extend(["--device=%s" % self.device])
```
Comment (michelwi) [inline returnn/compile.py:103]: Wrong amount of whitespace. 
Also why extend with a single item list when there is append directly below?

**Pull request merged**